### DISCUSSION TITLE
API Token for Groups

### DISCUSSION THREAD

**Discussion by remchuk** [netbox-community/netbox#12806] [Q&A]
Hey ! So I want to create groups for users and assign an API token for each group.

I want to make a way to send a header using a webhook of the group. 

For my understanding this is not a feature but is there a way to implement it ?

**Comment by candlerb** [comment]
> So I want to create groups for users and assign an API token for each group.

Netbox API tokens belong to users, not to groups.  You would have to assign an API token for each user.

> I want to make a way to send a header using a webhook of the group.

That sounds like a completely different question. API tokens are for external services making *inbound* requests to Netbox. Webhooks are for Netbox making *outbound* requests to other systems.  Webhooks are nothing to do with groups (or indeed users).

But in a webhook you can add custom headers to the request generated by Netbox (which might include an API token to a remote system).  And I believe you can do jinja2 template expansion to make it dynamic.

**Reply by remchuk** [reply]
But lets say I have 3 teams in my company and I want to give them different permissions based on groups
 
or another case scenario, I have a script that I want to separate based on the group of the person making a change. Lets say only if a member of team A makes changes to lets say switches then allow it.

**Reply by candlerb** [reply]
> But lets say I have 3 teams in my company and I want to give them different permissions based on groups

No problem. You make three groups, assign permissions to the groups, and you add the team members into the groups.

API tokens are still per-user.  When a user accesses Netbox with their personal API token, they get the permissions from the group(s) they are members of.

That is: regardless of whether a user logs into the web UI using their username/password, or talks to the API using their API token, they get exactly the same set of permissions enforced.

> I have a script that I want to separate based on the group of the person making a change. Lets say only if a member of team A makes changes to lets say switches then allow it.

When you say "script", do you mean a [Netbox Custom Script](https://docs.netbox.dev/en/stable/customization/custom-scripts/) that runs within Netbox, or an external script which talks to Netbox via the REST API, or talks to the Django ORM?
